ci/build.sh

```diff
@@ -82,43 +115,34 @@ if [ -n "$GITHUB_TOKEN" ] && [ "$TRAVIS_EVENT_TYPE" == "pull_request" ] && [[ !
       exit 1
     fi
   fi
-fi
+}
 
-# Test if template files are generated correctly for various values.yaml
-echo "Test helm templates generation"
-if ./tests/run.sh; then
-  echo "Helm templates generation test passed"
-else
-  echo "Tracing templates generation test failed"
-  exit 1
-fi
+function build_docker_image() {
+  local tag
+  local no_cache
+  tag="${1}"
 
-# Test upgrade script
-echo "Test upgrade script..."
-if ./tests/upgrade_script/run.sh; then
-  echo "Upgrade Script test passed"
-else
-  echo "Upgrade Script test failed"
-  exit 1
-fi
+  echo "Building docker image with ${tag}:local in $(pwd)..."
+  cd ./deploy/docker || exit 1
+  no_cache="--no-cache"
+  if [[ "$DOCKER_USE_CACHE" == "true" ]]; then
+    no_cache=""
+  fi
+  docker build . -f ./Dockerfile -t "${tag}:local" ${no_cache:+"--no-cache"}
+  rm -f ./gems/*.gem
+  cd ../.. || exit 1
```

Comment:
Maybe use a subshell or even better `pushd`+`popd`?
I'm not a huge fan of relative paths like `../..` in scripts

Reply:
Will go with `pushd` + `popd`